src/main/drivers/bus_i2c.h

```diff
@@ -64,9 +80,20 @@ typedef struct i2cState_s {
     volatile uint8_t* read_p;
 } i2cState_t;
 
+typedef struct i2cPinConfig_s {
+    ioTag_t ioTagSCL[I2CDEV_MAX];
+    ioTag_t ioTagSDA[I2CDEV_MAX];
+} i2cPinConfig_t;
+
 void i2cInit(I2CDevice device);
 bool i2cWriteBuffer(I2CDevice device, uint8_t addr_, uint8_t reg_, uint8_t len_, uint8_t *data);
 bool i2cWrite(I2CDevice device, uint8_t addr_, uint8_t reg, uint8_t data);
 bool i2cRead(I2CDevice device, uint8_t addr_, uint8_t reg, uint8_t len, uint8_t* buf);
 
 uint16_t i2cGetErrorCounter(void);
+void i2cInitAll();
+void i2cPinConfigDefault(void);
+void i2cPinConfigSet(int bus, ioTag_t scl, ioTag_t sda);
+
+extern i2cDevice_t i2cHardwareMap[I2CDEV_MAX];
```

Comment:
Don't do this. Use function to get actual item count(defined in same file as i2cHardwareMap) , to hide implementation details from `bus_i2c.h`. Compiler with LTO will optimize it away, so there will be no overhead.

Also you should probably use different name - `i2cHardwareMap` got renamed to `i2cHardwareConfig`, so it is better when compiler can catch instances that were forgotten. This may cause nasty bugs when merging.